infoblox/resourceIPAllocation.go

```diff
@@ -21,23 +21,41 @@ func resourceIPAllocation() *schema.Resource {
 				DefaultFunc: schema.EnvDefaultFunc("networkViewName", "default"),
 				Description: "Network view name available in Nios server.",
 			},
-			"host_name": &schema.Schema{
+			"vm_name": &schema.Schema{
 				Type:        schema.TypeString,
 				Required:    true,
 				DefaultFunc: schema.EnvDefaultFunc("hostName", nil),
-				Description: "The name of the record.",
+				Description: "The name of the VM.",
 			},
 			"cidr": &schema.Schema{
 				Type:        schema.TypeString,
 				Required:    true,
 				DefaultFunc: schema.EnvDefaultFunc("net_address", nil),
 				Description: "Give the address in cidr format.",
 			},
+			"zone": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("zone", nil),
+				Description: "Zone under which record has to be created .",
+			},
+			"enable_dns": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("enable_dns", nil),
+				Description: "flag that defines if the host reocrd is used for DNS or IPAM Puurposes .",
+			},
+			"dns_view": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("dns_view", nil),
+				Description: "Dns View under which the zone has been created .",
+			},
 			"ip_addr": &schema.Schema{
 				Type:        schema.TypeString,
 				Optional:    true,
 				DefaultFunc: schema.EnvDefaultFunc("ipaddr", nil),
-				Description: "IP address of your instance in cloud.",
+				Description: "IP address you want to allocate yourinstance in cloud.If field is not specified , it akes next avaliable ip address",
```

Comment:
punctuation,
IP address of your instance in cloud.  For static set ip_addr for dynamic leave this empty.